### ISSUE TITLE
Tx from mempool not included in DEV mode

### ISSUE THREAD

**Issue by mm-zk** [paradigmxyz/reth#6606]
### Describe the bug

When transaction ends up in the mempool (for example as it came with a higher nonce), it will stay there forever in the 'dev' mode. 
The expected behavior is that it should be included to the chain, when possible.

### Steps to reproduce

**Start a node in DEV mode**
```
./reth  node --http  -vvvvv --dev --instance 1 --dev.block-max-transactions 1 --datadir /tmp/18.6
```
Send transaction with nonce 1 - that will go to mempool
```
cast send -r http://localhost:8545 0x8a91dc2d28b689474298d91899f0c1baf62cb85b --value 1ether  --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --nonce 1 --async 
> 0x2137698178d3c9e69730d262dd67918b3220d18b8255073c247508e6117b6dab
```
Send transaction with nonce 0 - that will be executed:
```
cast send -r http://localhost:8545 0x8a91dc2d28b689474298d91899f0c1baf62cb85b --value 1ether  --private-key 0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80 --nonce 0 --async 
> 0xc722125a67295fcd8e4f1ad9bbf180881f62129bdd05aa9d9cf7de7efba823bb
```

The first transaction (with nonce 1) - will stay with blockhash null (in mempool) forever:

```
curl -X POST \     -H "Content-Type: application/json" \                   
     --data '{"jsonrpc":"2.0","method":"eth_getTransactionByHash","params":["0x2137698178d3c9e69730d262dd67918b3220d18b8255073c247508e6117b6dab"],"id":1}' \
     http://localhost:854
```

```json
{"jsonrpc":"2.0","result":
{"hash":"0x2137698178d3c9e69730d262dd67918b3220d18b8255073c247508e6117b6dab",
"nonce":"0x1",
"blockHash":null,
"blockNumber":null,
"transactionIndex":null,
"from":"0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266","to":"0x8a91dc2d28b689474298d91899f0c1baf62cb85b","value":"0xde0b6b3a7640000","gasPrice":"0x12a05f200","gas":"0x5208","maxFeePerGas":"0x12a05f200","maxPriorityFeePerGas":"0xb2d05e00","input":"0x","r":"0x73653ca3ae5329f5221c221644ebdb3cf26d3b1580e08a7426f015367365f0f3","s":"0x58c4642b85de7e257239248c40987b28aa5e6d99b7097ce3b7df45e4bd757529","v":"0x1","yParity":"0x1","chainId":"0x539","accessList":[],"type":"0x2"},"id":1}
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

reth Version: 0.1.0-alpha.18

### What database version are you on?

Current database version: 1
Local database version: 1

### What type of node are you running?

Archive (default)

### Code of Conduct

- [X] I agree to follow the Code of Conduct

**Comment by mm-zk** [comment]
TL;DR  - what should happen to transactions that were just included in a newly minted block, that wasn't 'committed' yet.


I looked a little bit more at this, and the problem happens due to the deletion/pruning logic - especially when node it generating the block.

There are two methods 'delete' (that just removes the transaction from the pool) and 'prune' - that also 'moves the next nonce transaction to the list of 'independent transactions'.

Now, when the node creates the block, it immediately removes the transactions that were included there (but doesn't prune them yet). The pruning should happen only once it sees that the block was successfully created. Unfortunately the second step fails - as the transactions are already gone (so the node no longer has the mapping between the transaction id and user/nonce).

And that's how the second transaction is now stuck in the Pending forever.

The short hack (at least for the 'dev' setup) - would be to call 'prune' instead of 'delete' when transaction is added to the block. But I guess that the real solution would be to have a separate SubPool for such transactions?

**Comment by mattsse** [comment]
thanks for the repro, will investigate

**Comment by mattsse** [comment]
this can be closed now, right?
